substrate/frame/support/src/traits/tokens/fungible/regular.rs

```diff
@@ -213,7 +213,10 @@ pub trait Unbalanced<AccountId>: Inspect<AccountId> {
 		} else {
 			old_balance.checked_add(&amount).ok_or(ArithmeticError::Overflow)?
 		};
-		if new_balance < Self::minimum_balance() {
+
+		// look at total balance to ensure ED is respected.
+		let new_total_balance = amount.saturating_add(Self::total_balance(who));
```

Comment:
The holds intentionally was not included into ED, I think the reason is that with the new fungibles contract we wanna guaranty that holds are always slashable, for the same reason we introduced the hold reasons. To change this you need really strong reasoning and for sure a separate PR.

Reply:
> a separate PR

A separate PR is fair.

> What is your explanation of this change

I wanted to retain the current behaviour that any staker currently can stake all their balance including ED. Also, ED requirement even when an account has an active hold seemed like an oversight to me, since the account can/should not be dusted if it has an active hold.

>  I think the reason is that with the new fungibles contract we wanna guaranty that holds are always slashable

How is it related? Why would holds not be slashable if free part of the balance does not have ED requirements on its own? All held amount can still be slashed and account would be dusted in that case right? Or am I missing something.

Reply:
what if account has some consumer or provider reference? not staking related